src/config/SourcegraphConfigOptions.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+final class PhabricatorSourcegraphConfigOptions extends PhabricatorApplicationConfigOptions
+{
+    public function getName()
+    {
+        return pht('Sourcegraph');
+    }
+
+    public function getDescription()
+    {
+        return pht('Configure Sourcegraph Settings.');
+    }
+
+    public function getIcon()
+    {
+        return 'fa-asterisk';
+    }
+
+    public function getGroup()
+    {
+        return 'apps';
+    }
+
+    public function getOptions()
+    {
+        $repo_type = 'sourcegraph.repo';
+        $repo_example = array(
+            array(
+                'path' => 'gorilla/mux',
+                'callsign' => 'MUX',
+            ),
+        );
+
+        $repo_example = id(new PhutilJSON())->encodeAsList(
+            $repo_example);
+
+        return array(
+            $this->newOption(
+                'sourcegraph.url',
+                'string',
+                null)
+                ->setDescription(pht('URL to Sourcegraph.'))
+                ->addExample('https://sourcegraph.example.com', pht('Valid Setting')),
+            $this->newOption('sourcegraph.repos', $repo_type, array())
+                ->setDescription(pht(
+                    'This option is only required for repositories defined in `%s` array found in your Sourcegraph site admin configuration.' .
+                    "\n\n" .
+                    'The object is an array with all elements of the type object. The array object has the following properties:' .
+                    "\n\n" .
+                    '`%s` (string, required) Display path for the url e.g. `%s`' .
```

Comment:
- "Display path" sounds like it's referring to a new concept that is different from "path" (what does "display" mean?).
- Also, it's "for the repository" not "for the URL".
- No need to give an example here, since the same example is repeated right below in the "Simple Example"

So what about just:

```
`%s` (string, required) The path of the repository on Sourcegraph
```

(BTW, "URL" should be capitalized; see https://sourcegraph.sgdev.org/github.com/sourcegraph/website/-/blob/STYLEGUIDE.md#examples.)